ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveRelFieldTrimmer.java

```diff
@@ -555,7 +556,8 @@ private Aggregate rewriteGBConstantKeys(Aggregate aggregate, ImmutableBitSet fie
     RexBuilder rexBuilder = aggregate.getCluster().getRexBuilder();
     final List<RexNode> newProjects = new ArrayList<>();
 
-    final List<RexNode> inputExprs = input.getChildExps();
+    final List<RexNode> inputExprs = new ArrayList<>();
+    input.accept(new ChildExpsRexShuttle(inputExprs));
     if (inputExprs == null || inputExprs.isEmpty()) {
       return aggregate;
     }
```

Comment:
This is comment is not related with the changes but it is a bit more general. The intention of this piece of code seems to be to find if all group by expressions are constant. However, I am not sure if `getChildExps()` is the right way to go. Most likely the code in `AggregateProjectPullUpConstantsRule` is more appropriate for this.